From 7eb63702d34ae17367494deda0b414cadc152f2a Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsht@xilinx.com>
Date: Wed, 21 Sep 2016 17:04:48 +0530
Subject: [PATCH 05/24] Fix Invalid type error for h265enc

- This fixes invalid type error for h265 enc element
  shown while running "gst-inspect-1.0 omx".

- Declare get_type function for h265 enc element and
  fix compilation for generating .lo file for h265 enc
  element.

Signed-off-by: Devarsh Thakkar <devarsht@xilinx.com>
Upstream Status: Pending
---
 omx/Makefile.am | 8 ++++----
 omx/gstomx.c    | 4 +++-
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/omx/Makefile.am b/omx/Makefile.am
index 1492ad1..80e3c92 100644
--- a/omx/Makefile.am
+++ b/omx/Makefile.am
@@ -32,8 +32,7 @@ libgstomx_la_SOURCES = \
 	$(VP8_C_FILES) \
 	$(THEORA_C_FILES) \
 	gstomxmpeg4videoenc.c \
-	$(H265_C_FILES) \
-	$(H265_H_FILES) \
+	gstomxh265enc.c \
 	gstomxh264enc.c \
 	gstomxh263enc.c \
 	gstomxaacdec.c \
@@ -41,7 +40,7 @@ libgstomx_la_SOURCES = \
 	gstomxaacenc.c \
 	gstomxaudiosink.c \
 	gstomxanalogaudiosink.c \
-	gstomxhdmiaudiosink.c	
+	gstomxhdmiaudiosink.c
 
 noinst_HEADERS = \
 	gstomx.h \
@@ -60,6 +59,7 @@ noinst_HEADERS = \
 	$(VP8_H_FILES) \
 	$(THEORA_H_FILES) \
 	gstomxmpeg4videoenc.h \
+	gstomxh265enc.h \
 	gstomxh264enc.h \
 	gstomxh263enc.h \
 	gstomxaacdec.h \
@@ -67,7 +67,7 @@ noinst_HEADERS = \
 	gstomxaacenc.h \
 	gstomxaudiosink.h \
 	gstomxanalogaudiosink.h \
-	gstomxhdmiaudiosink.h 	
+	gstomxhdmiaudiosink.h
 
 if !HAVE_EXTERNAL_OMX
 OMX_INCLUDEPATH = -I$(abs_srcdir)/openmax
diff --git a/omx/gstomx.c b/omx/gstomx.c
index 9a77e5b..8e2cb61 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -32,12 +32,14 @@
 #include "gstomxmpeg2videodec.h"
 #include "gstomxmpeg4videodec.h"
 #include "gstomxh264dec.h"
+#include "gstomxh265dec.h"
 #include "gstomxh263dec.h"
 #include "gstomxvp8dec.h"
 #include "gstomxtheoradec.h"
 #include "gstomxwmvdec.h"
 #include "gstomxmpeg4videoenc.h"
 #include "gstomxh264enc.h"
+#include "gstomxh265enc.h"
 #include "gstomxh263enc.h"
 #include "gstomxaacdec.h"
 #include "gstomxmp3dec.h"
@@ -2246,7 +2248,7 @@ static const GGetTypeFunction types[] = {
   gst_omx_h264_dec_get_type, gst_omx_h263_dec_get_type,
   gst_omx_wmv_dec_get_type, gst_omx_mpeg4_video_enc_get_type,
   gst_omx_h264_enc_get_type, gst_omx_h263_enc_get_type,
-  gst_omx_aac_enc_get_type, gst_omx_mjpeg_dec_get_type,
+  gst_omx_h265_enc_get_type, gst_omx_aac_enc_get_type, gst_omx_mjpeg_dec_get_type,
   gst_omx_aac_dec_get_type, gst_omx_mp3_dec_get_type
 #ifdef HAVE_VP8
       , gst_omx_vp8_dec_get_type
-- 
2.7.4

